### ISSUE TITLE
[Request] Filtering Mid-Stream

### ISSUE THREAD

**Issue by JoshuaDoes** [jonas747/dca#6]
Studying the source of `dca` and its current setup, I've been doing my best to try and research if there's any possible way to filter audio output mid-stream using FFMPEG and stdin. My reason for trying to figure this out is I absolutely love the simplistic library `dca` provides (and I even plan on adding a few things of my own in the near future with a pull request), however it seems almost impossible to change the volume mid-stream or apply equalizer effects mid-stream unless FFMPEG specifically allows it.

If you have the knowledge on how to do this, it'd be an amazing feature for `dca` to have. Even if you don't feel like doing it yourself, I'd be happy to help add it for things like changing the volume mid-stream should you or anyone else know how to go about it.

Thanks in advance for anything pertaining this,
JoshuaDoes

**Comment by alexwh** [comment]
It's possible, but you need to use an obscure sparsely documented option: https://ffmpeg.org/ffmpeg-filters.html#zmq_002c-azmq

Outside of this document and a few mailing list conversations, I can't find much more information about doing this on the fly without restarting ffmpeg (not ideal).

**Comment by alexwh** [comment]
I've also been wanting for a little more control over the audio stream data and ffmpeg options. Personally, I think restarting a new ffmpeg process with filters applied is a better solution than running two in tandem (you could keep the old one running until the new one is ready to send data, and sync their playing time, then make the switch).

Without doing that though, zmq seems to be the only supported way of doing this within ffmpeg, next is just figuring out the arcane filter syntax ffmpeg uses. I believe if the process is initialized with all the filters you will ever want to use, but with their parameters set to a default value, it is then possible to tweak their numbers with zmq messages.

**Comment by JoshuaDoes** [comment]
If you can figure out a way to do this purely in Go, please let me know!

**Comment by jonas747** [comment]
Ah i didn't know ffmpeg had that kinda feature, the documentation on this seems really shit tough (which is probably why i didn't know about it)

**Comment by JoshuaDoes** [comment]
Yeah I've looked into it myself and it's not fun to work with due to the lack of proper documentation.
